x-pack/plugins/security_solution/public/common/components/agents/agent_status/endpoint/endpoint_agent_status.tsx

```diff
@@ -0,0 +1,170 @@
+    const { data: endpointPendingActions } = useGetEndpointPendingActionsSummary(
+      [endpointHostInfo.metadata.agent.id],
+      {
+        refetchInterval: autoRefresh ? DEFAULT_POLL_INTERVAL : false,
+        enabled: !pendingActions,
+      }
+    );
+
+    const [hasPendingActions, hostPendingActions] = useMemo<
+      [boolean, EndpointPendingActions['pending_actions']]
+    >(() => {
+      if (!endpointPendingActions && !pendingActions) {
+        return [false, {}];
+      }
+
+      const pending = pendingActions
+        ? pendingActions
+        : endpointPendingActions?.data[0].pending_actions ?? {};
+
+      return [Object.keys(pending).length > 0, pending];
+    }, [endpointPendingActions, pendingActions]);
+
+    const status = endpointHostInfo.host_status;
+    const isIsolated = Boolean(endpointHostInfo.metadata.Endpoint.state?.isolation);
+
+    return (
+      <EuiFlexGroupStyled
+        gutterSize="none"
+        responsive={false}
+        className="eui-textTruncate"
+        data-test-subj={dataTestSubj}
+      >
+        <EuiFlexItem grow={false}>
+          <EuiBadge
+            color={status != null ? HOST_STATUS_TO_BADGE_COLOR[status] : 'warning'}
+            data-test-subj={getTestId('agentStatus')}
+            className="eui-textTruncate"
+          >
+            {getAgentStatusText(status)}
+          </EuiBadge>
+        </EuiFlexItem>
+        {(isIsolated || hasPendingActions) && (
+          <EuiFlexItem grow={false} className="eui-textTruncate isolation-status">
+            <AgentResponseActionsStatus
+              data-test-subj={getTestId('actionStatuses')}
+              isIsolated={isIsolated}
+              pendingActions={hostPendingActions}
+            />
+          </EuiFlexItem>
+        )}
+      </EuiFlexGroupStyled>
+    );
+  }
+);
+EndpointAgentStatus.displayName = 'EndpointAgentStatus';
+
+export interface EndpointAgentStatusByIdProps {
+  endpointAgentId: string;
+  /**
+   * If set to `true` (Default), then the endpoint status and isolation/action counts will
+   * be kept up to date by querying the API periodically
+   */
+  autoRefresh?: boolean;
+  'data-test-subj'?: string;
+}
+
+/**
+ * Given an Endpoint Agent Id, it will make the necessary API calls and then display the agent
+ * status using the `<EndpointAgentStatus />` component.
+ *
+ * NOTE: if the `HostInfo` is already available, consider using `<EndpointAgentStatus/>` component
+ * instead in order to avoid duplicate API calls.
+ */
+export const EndpointAgentStatusById = memo<EndpointAgentStatusByIdProps>(
+  ({ endpointAgentId, autoRefresh, 'data-test-subj': dataTestSubj }) => {
+    const { data } = useGetEndpointDetails(endpointAgentId, {
+      refetchInterval: autoRefresh ? DEFAULT_POLL_INTERVAL : false,
+    });
+
+    const emptyValue = (
+      <EuiText size="xs" data-test-subj={dataTestSubj}>
+        <p>{getEmptyValue()}</p>
+      </EuiText>
+    );
+
+    if (!data) {
+      return emptyValue;
+    }
```

Comment:
NIT: This `emptyValue` variable does not seem necessary

```suggestion
    if (!data) {
      return (
        <EuiText size="xs" data-test-subj={dataTestSubj}>
          <p>{getEmptyValue()}</p>
        </EuiText>
      );
    }
```